tests/plugins/test_default_qubit_autograd.py

```diff
@@ -0,0 +1,341 @@
+
+        @qml.qnode(dev, diff_method=diff_method, interface="autograd")
+        def circuit(x, weights, w=None):
+            """In this example, a mixture of scalar
+            arguments, array arguments, and keyword arguments are used."""
+            qml.QubitStateVector(1j * np.array([1, -1]) / np.sqrt(2), wires=w)
+            operation(x, weights[0], weights[1], wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        # Check that the correct QNode type is being used.
+        if diff_method == "backprop":
+            assert isinstance(circuit, qml.qnodes.PassthruQNode)
+            assert not hasattr(circuit, "jacobian")
+        else:
+            assert not isinstance(circuit, qml.qnodes.PassthruQNode)
+            assert hasattr(circuit, "jacobian")
+
+        def cost(params):
+            """Perform some classical processing"""
+            return circuit(params[0], params[1:], w=0) ** 2
+
+        theta = 0.543
+        phi = -0.234
+        lam = 0.654
+
+        params = np.array([theta, phi, lam], requires_grad=True)
+
+        res = cost(params)
+        expected_cost = (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi)) ** 2
+        assert np.allclose(res, expected_cost, atol=tol, rtol=0)
+
+        res = qml.grad(cost)(params)
+        expected_grad = (
+            np.array(
+                [
+                    np.sin(theta) * np.cos(lam) * np.cos(phi),
+                    np.cos(theta) * np.cos(lam) * np.sin(phi) + np.sin(lam) * np.cos(phi),
+                    np.cos(theta) * np.sin(lam) * np.cos(phi) + np.cos(lam) * np.sin(phi),
+                ]
+            )
+            * 2
+            * (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi))
+        )
+        assert np.allclose(res, expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("interface", ["tf", "torch"])
+    def test_error_backprop_wrong_interface(self, interface, tol):
+        """Tests that an error is raised if diff_method='backprop' but not using
+        the Autograd interface"""
+        dev = qml.device("default.qubit.autograd", wires=1)
+
+        def circuit(x, w=None):
+            qml.RZ(x, wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        with pytest.raises(
+            ValueError,
+            match="default.qubit.autograd only supports diff_method='backprop' when using the autograd interface",
+        ):
+            qml.qnode(dev, diff_method="backprop", interface=interface)(circuit)
+
+
+class TestHighLevelIntegration:
+    """Tests for integration with higher level components of PennyLane."""
+
+    @pytest.mark.xfail(
+        strict=True,
+        reason="qml.broadcast currently performs input type validation, only allowing Nones or Iterables. "
+        "The Autograd Arraybox is not an iterable, so an error is raised. This can be avoided by "
+        "simply modifying this type validation.",
+    )
+    def test_qnode_collection_integration(self):

```

Comment:
Does this mean we have no tests to make sure it plays nicely with QNodeCollections?

Reply:
I'll add some in :+1:

@mariaschuld mentioned it should be okay to simply remove this input check in the template package

Reply:
Done!